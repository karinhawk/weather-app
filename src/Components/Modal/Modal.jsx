import "./Modal.scss"
import cross from "../../assets/cross.svg"

const Modal = ({ toggleModal, handleUnit }) => {


    return (
        <div className='modal'>
            <div className="modal__floater">
                <div className="modal__content">
                <img onClick={toggleModal} className="modal__cross" src={cross} alt="" />
                    <div className="modal__options">
                    <h2 className="modal__options__heading">Choose Celsius or Fahrenheit</h2>
                    <div className="modal__options__div">
                    <div className="modal__options__div__c">
                    <input className="modal__options__div__c__checkbox" type="checkbox" id="c" onClick={handleUnit} />
                        <label className="modal__options__div__c__label" htmlFor="toggle_switch">Celsius</label>                       
                    </div>
                    <div className="modal__options__div__f">
                    <input className="modal__options__div__f__checkbox" type="checkbox" id="f" onClick={handleUnit}/>
                        <label className="modal__options__div__f__label" htmlFor="toggle_switch">Fahrenheit</label>
                    </div>
                    </div>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Modal