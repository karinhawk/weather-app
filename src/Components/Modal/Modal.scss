.modal {
    height: 90vh;
    width: 100vw;
    background-color: rgba(black, 0.5);
    display: flex;
    justify-content: center;
    align-items: center;
    position: fixed;
    top: 55%;
    left: 50%;
    transform: translate(-50%, -50%);

    &__floater {
        position: relative;
        background-color: rgb(208, 231, 222);
        border: 0.5rem solid rgb(147, 189, 193);
        width: 80vw;
        height: 25vh;
        border-radius: 0.5rem;
        display: flex;
        justify-content: center;
        align-items: center;

    }

    &__cross {
        position: relative;
        height: 2rem;
        width: 2rem;
        top: 10%;
        left: 125%;

        &:hover {
            cursor: pointer;
        }
    }

    &__content {
        height: 25vh;
        width: 50vw;
        position: relative;
        left: -10%;
        bottom: 10%;
        padding: 1rem;
        display: flex;
        flex-direction: column;
        justify-content: space-around;
        color: black;
    }
    &__options {
        height: 20vh;
        width: 60vw;
        position: relative;
        display: flex;
        flex-direction: column;
        justify-content: space-around;

        &__questions {
            height: 12vh;
            display: flex;
            flex-direction: column;
            justify-content: space-between;
        }

        &__heading {
            font-size: 1.5rem;
            font-weight: 500;
            text-decoration: 3px rgb(60, 115, 171) wavy underline;
        }

        &__prompt {
            font-size: 1rem;
            font-weight: 500;
            text-decoration: 2px rgb(60, 115, 171) wavy underline;
        }
        &__div {
            height: 8vh;
            display: flex;
            flex-direction: column;
            justify-content: space-between;

            &__c {
                width: 20vw;
                display: flex;
                justify-content: space-between;
            }
            &__f {
                width: 27vw;
                display: flex;
                justify-content: space-between;
            }
        }
    }
}


@media screen and (min-width: 720px) {

    .modal {
    &__floater {
        height: 40vh;
        width: 25vw;
    }

    &__cross {
        height: 2.5rem;
        width: 2.5rem;
        top: -10%;
        left: 90%;
    }

    &__content {
        height: 30vh;
        width: 22vw;
        left: 5%;
        bottom: 5%;
    }
    &__options {
        height: 25vh;
        width: 60vw;

        &__questions {
            height: 12vh;
        }

        &__heading {
            font-size: 2rem;
            font-weight: 500;
        }
        &__prompt {
            font-size: 1.3rem;
        }
        &__div {
            height: 6vh;

            &__c {
                width: 4vw;
            }
            &__f {
                width: 5.5vw;
            }
        }
    }


}

}