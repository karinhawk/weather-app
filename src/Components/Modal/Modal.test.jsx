import Modal from "./Modal.jsx"
import { getByText, render, screen } from "@testing-library/react";

it("Should render the modal when activated", () => {
    const {asFragment} = render(<Modal/>)
    expect(asFragment(<Modal />)).toMatchSnapshot()
  });

it("Should render the text in the modal", () => {
  render(<Modal />)
  const text = screen.getByText("OPTIONS")
  expect(text).toBeInTheDocument()
});