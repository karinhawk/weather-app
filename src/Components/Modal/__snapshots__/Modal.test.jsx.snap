// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the modal when activated 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modal__floater"
    >
      <div
        class="modal__content"
      >
        <img
          alt=""
          class="modal__cross"
          src="cross.svg"
        />
        <div
          class="modal__options"
        >
          <h1
            class="modal__options__heading"
          >
            OPTIONS
          </h1>
          <div
            class="modal__options__questions"
          >
            <h2
              class="modal__options__prompt"
            >
              Choose Celsius or Fahrenheit
            </h2>
            <div
              class="modal__options__div"
            >
              <div
                class="modal__options__div__c"
              >
                <input
                  class="modal__options__div__c__checkbox"
                  id="c"
                  type="checkbox"
                />
                <label
                  class="modal__options__div__c__label"
                  for="toggle_switch"
                >
                  Celsius
                </label>
              </div>
              <div
                class="modal__options__div__f"
              >
                <input
                  class="modal__options__div__f__checkbox"
                  id="f"
                  type="checkbox"
                />
                <label
                  class="modal__options__div__f__label"
                  for="toggle_switch"
                >
                  Fahrenheit
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
